### PR TITLE
Add Brotli Support

### PR DESCRIPTION
**Pull request by pmouawad**
Fix for:
https://issues.apache.org/jira/browse/COMPRESS-392

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -68,6 +68,12 @@ jar, tar, zip, dump, 7z, arj.
       <scope>test</scope>
     </dependency>
     <dependency>
+      <groupId>org.brotli</groupId>
+      <artifactId>dec</artifactId>
+      <version>0.1.1</version>
+      <scope>compile</scope>
```
Comment (bindul) [inline pom.xml]: Shouldn't this be an optional dependency, just like the xz one?

File: pom.xml [hunk 2]
```diff
@@ -85,6 +91,13 @@ jar, tar, zip, dump, 7z, arj.
       <version>${powermock.version}</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>commons-codec</groupId>
+      <artifactId>commons-codec</artifactId>
+      <version>1.10</version>
+      <scope>test</scope>
+    </dependency>
+
```
Comment (bodewig) [inline pom.xml]: I'd prefer encoding things by hand rather than adding a new dependency just for tests.

**Comment by bodewig** [comment]
Many thanks Philippe

see my comments on the test.

Also I haven't checked whether the brotli lib is an OSGi bundle, we may need to update the bundle-plugin configuration.

File: pom.xml (hunk 1 above)
Comment (bodewig) [inline pom.xml]: +1

**Comment by garydgregory** [comment]
Thank you for your patch. The only thing missing at first glance is a patch to `org.apache.commons.compress.compressors.CompressorStreamFactory`.

**Pull request closed without merging**